### PR TITLE
fix 'Queue empty' exception with scrolling in the break point conditi…

### PR DESCRIPTION
**Pull request by JieCarolHu**
…on completion window

Fixing issue https://github.com/dotnet/roslyn/issues/23891
<details><summary>Ask Mode template not completed</summary>

### Customer scenario

What does the customer do to get into this situation, and why do we think this
is common enough to address for this release.  (Granted, sometimes this will be
obvious "Open project, VS crashes" but in general, I need to understand how
common a scenario is)

### Bugs this fixes

(either VSO or GitHub links)

### Workarounds, if any

Also, why we think they are insufficient for RC vs. RC2, RC3, or RTW

### Risk

This is generally a measure our how central the affected code is to adjacent
scenarios and thus how likely your fix is to destabilize a broader area of code

### Performance impact

(with a brief justification for that assessment (e.g. "Low perf impact because no extra allocations/no complexity changes" vs. "Low")

### Root cause analysis

How did we miss it?  What tests are we adding to guard against it in the future?

### How was the bug found?

(E.g. customer reported it vs. ad hoc testing)

### Test documentation updated?

If this is a new non-compiler feature or a significant improvement to an existing feature, update https://github.com/dotnet/roslyn/wiki/Manual-Testing once you know which release it is targeting.

</details>

### CODE REVIEW

**Comment by JieCarolHu** [comment]
@dpoeschl Could you re-review this pls? Thanks
It was part of this PR: #26119

**Comment by JieCarolHu** [comment]
retest windows_debug_vs-integration_prtest

**Comment by JieCarolHu** [comment]
retest windows_release_vs-integration_prtest

**Comment by tmat** [comment]
@JieCarolHu Is the fix correct? My understanding is that the completion item span is empty when the completion has nothing to replace. That is, in most common scenario when you're type `foo.` and select one of the completion items. Should we treat this case the same as if nothing was selected?

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.cs [hunk 1]
```diff
@@ -140,8 +140,11 @@ internal override void OnModelUpdated(Model modelOpt)
             else
             {
                 var selectedItem = modelOpt.SelectedItemOpt;
-                var viewSpan = selectedItem == null ? (ViewTextSpan?)null : modelOpt.GetViewBufferSpan(selectedItem.Span);
-                var triggerSpan = viewSpan == null 
+                var viewSpan = selectedItem == null || selectedItem.Document == null
```
Comment (dpoeschl) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller.cs]: Can you add a comment to explain why the `selectedItem.Document` being `null` means that we can't use its `.Span` and what setting the `triggerSpan` to `null` will end up doing?

**Comment by dpoeschl** [comment]
@jinujoseph Since this is targeting `master` now, and we're planning to switch out the engine for 16.0, this fix will then no longer be relevant. We should keep the _scenario_ in mind for the new implementation though (perhaps leaving the bug open but putting it in the Editor Completion project for tracking). Does closing this and tracking the bug that way sound good to you?

**Comment by JieCarolHu** [comment]
retest windows_debug_vs-integration_prtest

**Comment by JieCarolHu** [comment]
Ignoring the failures on vs-integration_prtests, which has nothing to do with this change

**Pull request merged**